Frontend_Web_Development_React_Angular_Vue/React/Week 1/Getting_start_with_React.md

```diff
@@ -0,0 +1,149 @@
+
+Welcome to Node.js v<your-version-number>.
+Type ".help" for more information.
+
+## Create React App
+
+Step 2: You can now create a new React Application.
+
+In the same terminal,press CTRL+C twice ,type the following command and press enter to create your own blank new react app.
+
+```npx create-react-app my-react-app```
+
+This will create a new React Application.To make changes in the UI,you have to make changes in the ./src folder of your project.
+
+Step 3: To see your application running,type the following command in your terminal and press enter
+
+```cd my-react-app
+npm start```
+
+Step 4: You can now see the demo application running on **http://localhost:3000/** of your default browser.
+
+Now,let's make changes to transform this into a Hello World App.
+
+## Hello World Component with Stateless Functions
+
+Step 1:Create a folder called 'Components' in the ./src folder.
+
+These components can be easily imported in the ./App.js file and thereby rendered on localhost:3000/
+
+Step 2:Inside the ./Components create a new file called 'HelloWorld.jsx'
+
+Step 3:Put the following code in 'HelloWorld.jsx'
+
+````
+
+import React from 'react';
+
+const HelloWorld = () => {
+return (
+
+<div>
+<h1>Hello World</h1>
+</div>
+)
+}
+
+export default HelloWorld;
+
+```
+
+The most basic component in React is this kind of a stateless functional component.
+
+They do not require any kind of internal state management.
+
+You can treat them as simple Javascript Functions.
+This creates a stateless Hello World comnonent that renders a Hello World message on localhost:3000/.
+
+Step 4 :You can now easily import it to ./App.js.
+
+To do so,replace the code in ./App.js with the following:
+
+```
+
+import "./App.css";
+import HelloWorld from "./Components/HelloWorld";
+
+function App() {
+return (
+
+<div className="App">
+<HelloWorld/>
+</div>
+);
+}
+
+export default App;
+
+```
+
+You can make changes in the ./App.css to change the css.And whatever new or modified new class names you add to this file, can be used in ./App.js or any other file by importing it using
+
+`import "./App.css";`
+And connecting a particular class to an element using the following syntax.
+
+`<elementTag className="class"> 'Some Text' </elementTag>`
+
+## Absolute Basics of Creating Reusable Components
+
+React is known for the functionality of creating reuable components.And Props play a major role in it.Props are like attributes passed through these stateless functions to render different messages depending on these props.
+
+They still look the same without having to repeat the same code again and again.Let's convert our HelloWorld Component into a Reusable One.To do so,let's make some minor changes and add a prop to our HelloWorld Component.
+
+```
+
+import React from 'react';
+
+const HelloWorld = props => {
+return (
+
+<div>
+<h1>Hello {props.name}</h1>
+<h2>{props.content}</h2>
+</div>
+)
```

Comment:
Here also same mistake add **; (semicolon)** here

Reply:
Should I remove the props section totally and put some basic reusability?I have made all other changes